eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,129 @@
+      }
+    }
+  }
+
+  return $onboardedPackages
+}
+
+function Get-python-DocsMsTocData($packageMetadata, $docRepoLocation) {
+  # Fallback for package name
+  $packageLevelReadmeName = $packageMetadata.Package
+  if ($packageLevelReadmeName.StartsWith('azure-')) {
+    # Eliminate a preceeding "azure." in the readme filename.
+    # "azure-storage-blobs" -> "storage-blobs"
+    $packageLevelReadmeName = $packageLevelReadmeName.Substring(6)
+  }
+
+  if ($packageMetadata.PSObject.Members.Name -contains "FileMetadata") {
+    $readmeMetadata = &$GetDocsMsMetadataForPackageFn -PackageInfo $packageMetadata.FileMetadata
+    $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
+  }
+
+
+  $packageTocHeader = $packageMetadata.Package
+  if ($packageMetadata.DisplayName) {
+    $packageTocHeader = $packageMetadata.DisplayName
+  }
+  $output = [PSCustomObject]@{
+    PackageLevelReadmeHref = "~/docs-ref-services/{moniker}/$packageLevelReadmeName-readme.md"
+    PackageTocHeader       = $packageTocHeader
+    TocChildren            = @($packageMetadata.Package)
+  }
+
+  return $output
+}
+
+function Get-python-DocsMsTocChildrenForManagementPackages($packageMetadata, $docRepoLocation) {
+  return @($packageMetadata.Package)
+}
+
+function addManagementPackage($serviceEntry, $packageName) {
+  for ($i = 0; $i -lt $serviceEntry.items.Count; $i++) {
+    if ($serviceEntry.items[$i].name -eq "Management") {
+      $serviceEntry.items[$i].children += @($packageName)
+      return $serviceEntry
+    }
+  }
+
+  $serviceEntry.items += [PSCustomObject]@{
+    name  = "Management";
+    landingPageType = 'Service';
+    children = @($packageName)
+  }
+
+  return $serviceEntry
+}
+
+function Get-python-UpdatedDocsMsToc($toc) {
+  $services = $toc[0].items
+  for ($i = 0; $i -lt $services.Count; $i++) {
+
+    # Add "ADAL" to "Active Directory" service. This is onboarded through a repo
+    # source process that is not obvious in the CI configuration (no package
+    # name, only a repo URL)
+    if ($services[$i].name -eq 'Active Directory') {
+      $services[$i].items += [PSCustomObject]@{
+        name  = "adal";
+        children = @("adal")
+      }
+    }
+
+    # azure-mgmt-mixedreality is not onboarded in an obvious manner, it is
+    # onboarded using a URL to a specific dist.
+    if ($services[$i].name -eq 'Mixed Reality') {
+      $services[$i] = addManagementPackage $services[$i] 'azure-mgmt-mixedreality'
+    }
+
+    # Retired package
+    if ($services[$i].name -eq 'Container Registry') {
+      $services[$i] = addManagementPackage $services[$i] 'azure-mgmt-containerregistry'
+    }
+  }
+
+  $functionService =  [PSCustomObject]@{
+    name = 'Functions';
+    landingPageType = 'Service';
+    children = @('azure-functions', 'azure-functions-durable')
+  }
+
+  # The network service is not onboarded into the regular Python build because
+  # it takes many hours to build.
+  $networkService = [PSCustomObject]@{
```

Comment:
For these exceptions to the rule do you think it is easier to handle them in code or by putting them in a separate file that we just merge in? I'm not sure it which is easier but something to consider.